src/src/components/ObjectBrowser.jsx

```diff
@@ -3578,39 +3609,41 @@ class ObjectBrowser extends Component {
         return [];
     }
 
-    async _exportObjects(isAll, noStatesByExportImport) {
+    async _exportObjects(isAll, noStatesByExportImport, beautifyJsonExport, excludeSystemRepositoriesFromExport, excludeTranslations) {
         if (isAll) {
-            generateFile('allObjects.json', this.objects);
-        } else if (this.state.selected.length || this.state.selectedNonObject) {
-            const result = {};
-            const id = this.state.selected[0] || this.state.selectedNonObject;
-            const ids = this._getSelectedIdsForExport();
-
-            for (let i = 0; i < ids.length; i++) {
-                const key = ids[i];
-                result[key] = JSON.parse(JSON.stringify(this.objects[key]));
-
-                // read states values
-                if (result[key]?.type === 'state' && !noStatesByExportImport) {
-                    const state = await this.props.socket.getState(key);
-                    if (state) {
-                        result[key].val = state.val;
-                        result[key].ack = state.ack;
-                    }
+            generateFile('allObjects.json', this.objects, beautifyJsonExport, excludeSystemRepositoriesFromExport, excludeTranslations);
+            return;
+        }
+        if (!(this.state.selected.length || this.state.selectedNonObject)) {
+            window.alert(this.props.t('ra_Save of objects-tree is not possible'));
+            return;
+        }
+        const result = {};
+        const id = this.state.selected[0] || this.state.selectedNonObject;
+        const ids = this._getSelectedIdsForExport();
+
+        for (let i = 0; i < ids.length; i++) {
```

Comment:
```suggestion
        for (const key of ids) {
```

saves us one line and is easier to read